treeherder/perf/autoclassify.py

```diff
@@ -0,0 +1,296 @@
+import datetime
+import httplib
+import json
+
+from treeherder.model.derived import JobsModel
+from treeherder.perf.models import (PerformanceAlert,
+                                    PerformanceAlertSummary)
+from treeherder.webapp.api.utils import UrlQueryFilter
+
+treeherder_url = "treeherder.mozilla.org"
+hg_url = "hg.mozilla.org"
+request_count = 0
+
+
+def transform_alert(alert):
+    """
+    Transform the given Alert to the fromat required
+    """
+    return {"id": int(alert.id),
+            "signature": alert.series_signature.signature_hash}
+
+
+def transform_alertsummary(alertsummary):
+    """ Transform the given AlertSummary to the format required """
+    return {"id": int(alertsummary.id),
+            "result_set_id": int(alertsummary.result_set_id),
+            "prev_result_set_id": int(alertsummary.prev_result_set_id),
+            "last_updated": alertsummary.last_updated,
+            "repository": alertsummary.repository.name,
+            "alerts": [transform_alert(alert) for alert in alertsummary.alerts.all()],
+            "related_alerts": [transform_alert(alert) for alert in alertsummary.related_alerts.all()],
+            "status": int(alertsummary.status)}
+
+
+def transform_revision(revision):
+    """ Transform the given Revision to the format required """
+    return {"comments": revision["comments"],
+            "revision": revision["revision"]}
+
+
+def transform_resultset(resultset):
+    """ Transform the given Resultset to the format required """
+    return {"id": resultset["id"],
+            "comments": resultset["comments"],
+            "revision": resultset["revision"],
+            "revisions": [transform_revision(revision) for revision in resultset["revisions"]]}
+
+
+def treeherder_request(endpoint):
+    """Return a dictioanry of the JSON returned from an endpoint"""
+    conn = httplib.HTTPSConnection(treeherder_url)
+    headers = {"User-Agent": "Automatic downstream classification script"}
+    conn.request("GET", endpoint, {}, headers)
+
+    global request_count
+    request_count += 1
+    print "{}. {}".format(request_count, endpoint)
+
+    return json.loads(conn.getresponse().read())
+
+
+def hg_request(endpoint):
+    """Return a dictionary of the JSON returned from an endpoint"""
+    conn = httplib.HTTPSConnection(hg_url)
+    headers = {"User-Agent": "Automatic downstream classification script. If I'm overloading the server, please find royc on #treeherder"}
+    conn.request("GET", endpoint, {}, headers)
+
+    global request_count
+    request_count += 1
+    print "{}. {}".format(request_count, endpoint)
+
+    return json.loads(conn.getresponse().read())
+
+
+def get_all_alertsummaries(url="/api/performance/alertsummary/", pages=10):
+    """Returns a list of AlertSummary """
+    return [transform_alertsummary(summary) for summary in PerformanceAlertSummary.objects.order_by("-last_updated")[:100]]
```

Comment:
There is no need to paginate requests for something like this. I would also just put a function like this inline -- making it a seperate function just makes it harder to see what's really going on :)